developer-docs-site/docs/concepts/basics-gas-txn-fee.md

```diff
@@ -2,71 +2,122 @@
+:::tip Unit of gas
+👉 A **unit of gas** is a dimensionless number, expressed as an integer. The total gas units consumed by your transaction depends on the complexity of your transaction. The **gas price**, on the other hand, is expressed in terms of Aptos blockchain’s native coin (Octas). Also see [Transactions and States](/concepts/basics-txns-states) for how a transaction submitted to the Aptos blockchain looks like.
+:::
 
-This means that the minimum transaction fee, `MIN_TXN_FEE`, needs to be enough to cover the average cost of running the prologue and epilogue.
+## Gas and transaction fee on the Aptos blockchain
 
-After the prologue has run, and we've checked in part that the account can cover its gas liability, the rest of the transaction flow starts with the "gas tank" full at `max_gas_amount`. The `MIN_TXN_FEE` is charged, after which the gas tank is then charged for each instruction the VM executes. This per-instruction deduction continues until either:
-* The transaction execution is complete, after which the cost of storing the transaction data is charged, and the epilogue is run and the execution fee deducted, or
-* The "gas tank" becomes empty, in which case an `OutOfGas` error is raised.
+When a transaction is submitted to the Aptos blockchain, the transaction must contain the following mandatory gas fields:
 
-In the former, the fee is collected and the result of the transaction is persisted on the Aptos blockchain. In the latter, the execution of the transaction stops when the error is raised. After which, the total gas liability of the transaction is collected. No other remnants of the execution are committed other than the deduction in this case.
+- `max_gas_amount`: The maximum number of gas units that the transaction sender is willing to spend to execute the transaction. This determines the maximum computational resources that can be consumed by the transaction.
+- `gas_price`: The gas price the transaction sender is willing to pay. It is expressed in Octa units, where:
+    - 1 Octa = 10<sup>-8</sup> APT and 
+    - APT is the Aptos coin. 
+  
+  During the transaction execution, the total gas amount, expressed as:
+  ```
+  (total gas units consumed) * (gas_price)
+  ```
+  must not exceed `max_gas_amount`, or else the transaction will abort the execution.
 
-## Using gas to prioritize a transaction
+The transaction fee charged to the client will be at the most `gas_price * max_gas_amount`.
 
-When you send a transaction, it is prioritized based on different criteria. One of these is the normalized gas price for the transaction.
+## Gas parameters set by the governance
 
-For example:
+The following gas parameters are set by the Aptos governance. 
```

Comment:
It's very strange to say `by the Aptos governance` maybe `by Aptos governance` instead.